pallets/payment-streams/src/utils.rs

```diff
@@ -54,15 +55,15 @@ where
     ///
     /// Note: Maybe we should add a check to make sure the user has enough balance to pay for at least X amount of blocks?
     pub fn do_create_payment_stream(
-        bsp_account: &T::AccountId,
+        bsp_id: &ProviderIdFor<T>,
         user_account: &T::AccountId,
         rate: BalanceOf<T>,
     ) -> DispatchResult {
-        // Get the BSP ID of the BSP account
-        let bsp_id = <T::Providers as storage_hub_traits::ProvidersInterface>::get_provider(
-            bsp_account.clone(),
-        )
-        .ok_or(Error::<T>::NotABackupStorageProvider)?;
+        // Check that the given ID belongs to an actual provider
+        ensure!(
+            <T::ProvidersPallet as ProvidersInterface>::is_provider(*bsp_id),
```

Comment:
`is_provider` does not guarantee that the id is a BSP, it could be an MSP. I would also add a check using `is_bsp`. And would do this everywhere where we want to ensure the id is in fact a BSP.

Reply:
This was actually generic for a storage provider but the naming and docs did not reflect that. The latest commit now fixes that, ty 👍🏻